internal/pkg/cli/run_local_test.go

```diff
@@ -411,198 +398,71 @@ func TestRunLocalOpts_Execute(t *testing.T) {
-					close(stopCalled)
-					return errors.New("stop foo")
-				})
-				m.dockerEngine.EXPECT().Stop(gomock.Any(), expectedRunBarArgs.ContainerName).Return(nil)
-				m.dockerEngine.EXPECT().Rm(expectedRunBarArgs.ContainerName).Return(errors.New("rm bar"))
-				m.dockerEngine.EXPECT().Stop(gomock.Any(), expectedRunPauseArgs.ContainerName).Return(nil)
-				m.dockerEngine.EXPECT().Rm(expectedRunPauseArgs.ContainerName).Return(errors.New("rm stop"))
-			},
-			wantedError: fmt.Errorf("clean up %q: rm: rm bar\nclean up %q: stop: stop foo\nclean up %q: rm: rm stop", expectedRunBarArgs.ContainerName, expectedRunFooArgs.ContainerName, expectedRunPauseArgs.ContainerName),
-		},
-		"handles ctrl-c successfully, errors from running ignored": {
-			inputAppName:  testAppName,
-			inputWkldName: testWkldName,
-			inputEnvName:  testEnvName,
-			setupMocks: func(m *runLocalExecuteMocks) {
-				m.ecsLocalClient.EXPECT().TaskDefinition(testAppName, testEnvName, testWkldName).Return(taskDef, nil)
-				m.ssm.EXPECT().GetSecretValue(gomock.Any(), "mysecret").Return("secretvalue", nil)
-				m.ws.EXPECT().ReadWorkloadManifest(testWkldName).Return([]byte(""), nil)
-				m.interpolator.EXPECT().Interpolate("").Return("", nil)
-
-				runCalled := make(chan struct{})
-				stopCalled := make(chan struct{})
-				m.dockerEngine.EXPECT().Run(gomock.Any(), expectedRunPauseArgs).DoAndReturn(func(ctx context.Context, opts *dockerengine.RunOptions) error {
-					close(runCalled)
-					return nil
-				})
-				m.dockerEngine.EXPECT().IsContainerRunning(gomock.Any(), mockPauseContainerName).DoAndReturn(func(ctx context.Context, name string) (bool, error) {
-					<-runCalled
-					return true, nil
-				})
-				m.dockerEngine.EXPECT().Run(gomock.Any(), expectedRunFooArgs).DoAndReturn(func(ctx context.Context, opts *dockerengine.RunOptions) error {
-					syscall.Kill(syscall.Getpid(), syscall.SIGINT)
-					<-stopCalled
-					return errors.New("hi")
-				})
-				m.dockerEngine.EXPECT().Run(gomock.Any(), expectedRunBarArgs).Return(nil)
-
-				m.prog.EXPECT().Start(gomock.Any()).Return().Times(6)
-				m.prog.EXPECT().Stop(gomock.Any()).Return().Times(3)
-				m.dockerEngine.EXPECT().Stop(gomock.Any(), expectedRunFooArgs.ContainerName).DoAndReturn(func(ctx context.Context, id string) error {
-					close(stopCalled)
-					return nil
-				})
-				m.dockerEngine.EXPECT().Rm(expectedRunFooArgs.ContainerName).Return(nil)
-				m.dockerEngine.EXPECT().Stop(gomock.Any(), expectedRunBarArgs.ContainerName).Return(nil)
-				m.dockerEngine.EXPECT().Rm(expectedRunBarArgs.ContainerName).Return(nil)
-				m.dockerEngine.EXPECT().Stop(gomock.Any(), expectedRunPauseArgs.ContainerName).Return(nil)
-				m.dockerEngine.EXPECT().Rm(expectedRunPauseArgs.ContainerName).Return(nil)
+				}
+
+				count := 1
+				m.orchestrator.StopFn = func() {
+					switch count {
+					case 1:
+						errCh <- errors.New("error from stopping after sigint")
+					case 2:
```

Comment:
nice! I see we expect `runLocalOpts` to behaves like this:
1. Got SigInt, call `Stop()`.
2. Got err from `errCh`, call `Stop()` again.
3. Got err from `errCh`, call `Stop()` for the third time.

Reply:
that's right! i'd also add:

4. exit after `errCh` is closed

Reply:
:+1: this test case is awesome